src/service-api.ts

```diff
@@ -0,0 +1,326 @@
+
+        const result = await h5pValidator.validatePackage(contentFolder);
+        if (!result.isValid) {
+          throw new InvalidH5PFileError(result.error);
+        }
+
+        // try-catch block for remote storage cleanup
+        try {
+          // upload whole folder to public storage
+          await uploadH5P(targetFolder, remoteRootPath, member);
+
+          const item = await createH5PItem(
+            h5pFile.filename,
+            contentId,
+            remoteRootPath,
+            member,
+            parentId,
+          );
+          return item;
+        } catch (error) {
+          // delete public storage folder of this H5P if upload or creation fails
+          fileTaskManager.createDeleteFolderTask(member, {
+            folderPath: remoteRootPath,
+          });
+
+          // remove local temp folder, before rethrowing
+          rm(targetFolder, { recursive: true });
+
+          // log and rethrow to let fastify handle the error response
+          log.error('graasp-plugin-h5p: unexpected error occured while importing H5P:');
+          log.error(error);
+          throw error;
+        }
+        // end of try-catch block for remote storage cleanup
+      } finally {
+        // in all cases, remove local temp folder
+        rm(targetFolder, { recursive: true });
+      }
+      // end of try-catch block for local storage cleanup
+    },
+  );
+
+  /**
+   * Delete H5P assets on item delete
+   */
+  const deleteItemTaskName = itemTaskManager.getDeleteTaskName();
+  taskRunner.setTaskPostHookHandler<Item<H5PExtra>>(deleteItemTaskName, async (item, actor) => {
+    if (item.type !== H5P_ITEM_TYPE) {
+      return;
+    }
+    const deleteTask = fileTaskManager.createDeleteFolderTask(actor, {
+      folderPath: buildRootPath(item.extra.h5p.contentId),
+    });
+    await taskRunner.runSingle(deleteTask);
+  });
+
+  /**
+   * Copy H5P assets on item copy
+   */
+  const copyItemTaskName = itemTaskManager.getCopyTaskName();
+  taskRunner.setTaskPreHookHandler<Item<H5PExtra>>(copyItemTaskName, async (item, actor) => {
+    if (item.type !== H5P_ITEM_TYPE) {
+      return;
+    }
+    if (!item.extra?.h5p) {
+      throw new Error('Invalid state: missing previous H5P item extra on copy');
+    }
+
+    const contentId = v4();
+    const remoteRootPath = buildRootPath(contentId);
+    const copyTask = fileTaskManager.createCopyFolderTask(actor, {
+      originalFolderPath: buildRootPath(item.extra.h5p.contentId),
+      newFolderPath: remoteRootPath,
+    });
+    await taskRunner.runSingle(copyTask);
+
+    item.extra.h5p = {
+      contentId,
+      h5pFilePath: buildH5PPath(remoteRootPath, contentId),
+      contentFilePath: buildContentPath(remoteRootPath),
+    };
+  });
+
+  /**
+   * H5P assets proxy server
+   * Can be used to add access control to h5p content files
+   * With the current architecture, the H5P libraries control the
+   * fetching so we can't pass the server cookie
+   */
```

Comment:
I would like more details: so this code doesn't work because of the cookie? Is this code supposed to be used locally only? For test maybe?